compiler/rustc_codegen_ssa/src/base.rs

```diff
@@ -906,17 +906,22 @@ impl CrateInfo {
                 })
                 .collect();
             let prefix = if target.is_like_windows && target.arch == "x86" { "_" } else { "" };
+
+            // This loop only adds new items to values of the hash map, so the order in which they
+            // are added is not important.
+            #[allow(rustc::potential_query_instability)]
             info.linked_symbols
                 .iter_mut()
                 .filter(|(crate_type, _)| {
                     !matches!(crate_type, CrateType::Rlib | CrateType::Staticlib)
                 })
                 .for_each(|(_, linked_symbols)| {
-                    linked_symbols.extend(
-                        missing_weak_lang_items
-                            .iter()
-                            .map(|item| (format!("{prefix}{item}"), SymbolExportKind::Text)),
-                    );
+                    let mut symbols = missing_weak_lang_items
+                        .iter()
+                        .map(|item| (format!("{prefix}{item}"), SymbolExportKind::Text))
+                        .collect::<Vec<_>>();
+                    symbols.sort_unstable_by(|a, b| a.0.cmp(&b.0));
```

Comment:
This sort conflicts with the comment above, saying the order isn't important. Since `linked_symbols` is a hash map, the order really isn't important and the sort should be dropped.

Reply:
There are two sources of instability here, iteration over the outer hashmap, and then iterating over `missing_weak_lang_items`. Sadly, enabling `#[allow(rustc::potential_query_instability)]` hides the inner unstable iteration, which almost made me miss it! For the inner block of code, the order is important, since we put the things into a Vec. But for the outer iteration, it does not matter which Vec we fill first.

Reply:
Makes sense